adr/0002-use-containers-for-foreign-languages.md

```diff
@@ -0,0 +1,29 @@
+# 1. Use containers to deploy tools from foreign languages source code
+
+Date: 2017-10-25
+
+## Status
+
+Proposed
+
+## Context
+
+eLife has a small set of supported languages: PHP, Python, JavaScript; in-house developers are present for them. All other languages are defined as **foreign**.
+
+There are tools that are peculiar to our infrastructure, such as [goaws](https://github.com/p4tin/goaws) for AWS simulation, but are written in languages no one is an expert in (Go).
+
+There are also tools that were originally written in another language but are being adopted by us, like [INK](https://gitlab.coko.foundation/INK/ink-api) for document conversion, written in Ruby.
+
+These tools are usually distributed as source code. The operational overhead of writing formulas for the environment to build them is a form of waste.
+
+Some tools written in Java instead have a very stable runtime platform (Selenium, ElasticSearch, even Jenkins), as they are distributed as binaries.
```

Comment:
True, though Selenium has dependencies on Firefox versions (which was complicated to build as we're testing audio).

Reply:
I wonder if that could be done with a container at all, as it impacts the kernel modules. But long-term the way to expand it it to use BrowserStack and similar services, a different kind of outsourcing.

Reply:
Undocumented options like https://hub.docker.com/r/selenium/standalone-firefox/